metron-interface/metron-alerts/src/app/alerts/configure-rows/configure-rows.component.ts

```diff
@@ -16,53 +16,29 @@
  * limitations under the License.
  */
 import { Component, Input, HostListener, ElementRef, Output, EventEmitter } from '@angular/core';
-import { TableMetadata } from '../../model/table-metadata';
-import { ConfigureTableService } from '../../service/configure-table.service';
 
+export interface ConfigureRowsModel {
+  values: {
+    pageSize: number;
+    refreshInterval: number;
+  },
+  triggerQuery: boolean
+}
 @Component({
   selector: 'app-configure-rows',
   templateUrl: './configure-rows.component.html',
   styleUrls: ['./configure-rows.component.scss']
 })
 export class ConfigureRowsComponent  {
-
   showView = false;
-  tableMetadata = new TableMetadata();
 
   @Input() srcElement: HTMLElement;
-  @Output() sizeChange = new EventEmitter();
-  @Output() intervalChange = new EventEmitter();
-  @Output() configRowsChange = new EventEmitter();
-
-  constructor(private elementRef: ElementRef,
-              private configureTableService: ConfigureTableService) {}
-
-  @Input()
-  get size() {
-    return this.tableMetadata.size;
-  }
+  @Input() pageSize: number;
+  @Input() refreshInterval: number;
 
-  set size(val) {
-    this.tableMetadata.size = val;
-  }
+  @Output() configRowsChange = new EventEmitter<ConfigureRowsModel>();
 
-  @Input()
-  get interval() {
-    return this.tableMetadata.refreshInterval;
-  }
-
-  set interval(val) {
-    this.tableMetadata.refreshInterval = val;
-  }
-
-  @Input()
-  get tableMetaData() {
-    return this.tableMetadata;
-  }
-
-  set tableMetaData(val) {
```

Comment:
tableMetaData is no directly used by this component so I refactored it out.
Changes here was needed to separate config change events we like to trigger query from others.